File: NUnit/Version2/Saucery2/Capabilities/Base/BaseCapabilities.cs
using OpenQA.Selenium.Remote;
using Saucery2.Util;

namespace Saucery2.Capabilities.Base {
    internal abstract class BaseCapabilities {
        protected DesiredCapabilities Caps = null;
        private readonly string _testName;

        protected BaseCapabilities(string testName) {
            _testName = testName;
        }

        protected void AddSauceLabsCapabilities() {
            Caps.SetCapability(SauceryConstants.SAUCE_USERNAME_CAPABILITY, Enviro.SauceUserName);
            Caps.SetCapability(SauceryConstants.SAUCE_ACCESSKEY_CAPABILITY, Enviro.SauceApiKey);
            //This sets the Session column
            Caps.SetCapability(SauceryConstants.SAUCE_SESSIONNAME_CAPABILITY, _testName);
            //This sets the Build column
            Caps.SetCapability(SauceryConstants.SAUCE_BUILDNAME_CAPABILITY, Enviro.BuildNumber);
            //Improve performance on SauceLabs
            Caps.SetCapability(SauceryConstants.SAUCE_VUOP_CAPABILITY, false);
            //Caps.SetCapability(Constants.VISIBILITY_KEY, Constants.VISIBILITY_TEAM);
        }

        protected void AddSauceLabsCapabilities(string nativeApp) {
            AddSauceLabsCapabilities();
            if (nativeApp != null) {
                Caps.SetCapability(SauceryConstants.SAUCE_NATIVE_APP_CAPABILITY, nativeApp);
            }
        }

        public DesiredCapabilities GetCaps() {
            return Caps;
        }

        protected static string GetBrowser(string nativeApp, bool useChromeOnAndroid) {
            return nativeApp != null
                ? ""
                : useChromeOnAndroid
                    ? SauceryConstants.CHROME_BROWSER
                    : SauceryConstants.DEFAULT_ANDROID_BROWSER;
        }

        protected static string GetBrowser(string nativeApp) {
            return nativeApp != null ? "" : SauceryConstants.SAFARI_BROWSER;
        }
    }
}
/*
 * Copyright Andrew Gray, Full Circle Solutions
 * Date: 18th September 2014
 * 
 */

File: JUnit/saucery-core/src/main/java/au/com/fullcirclesolutions/saucery/capabilities/BaseCapabilities.java
package au.com.fullcirclesolutions.saucery.capabilities;

import au.com.fullcirclesolutions.saucery.utils.Enviro;
import au.com.fullcirclesolutions.saucery.utils.SauceryConstants;
import org.openqa.selenium.remote.DesiredCapabilities;

abstract class BaseCapabilities {
    DesiredCapabilities Caps = null;
    private final String _testName;

    BaseCapabilities(String testName) {
        _testName = testName;
    }

    void AddSauceLabsCapabilities() {
        //This sets the Session column
        Caps.setCapability(SauceryConstants.SAUCE_SESSIONNAME_CAPABILITY, _testName);
        //This sets the Build column
        Caps.setCapability(SauceryConstants.SAUCE_BUILDNAME_CAPABILITY, Enviro.BuildNumber());
        //Improve performance on SauceLabs
        Caps.setCapability(SauceryConstants.SAUCE_VUOP_CAPABILITY, false);
        //Caps.setCapability(Constants.VISIBILITY_KEY, Constants.VISIBILITY_TEAM);
    }

    void AddSauceLabsCapabilities(String nativeApp) {
        AddSauceLabsCapabilities();
        if (nativeApp != null) {
            Caps.setCapability(SauceryConstants.SAUCE_NATIVE_APP_CAPABILITY, nativeApp);
        }
    }

    DesiredCapabilities GetCaps() {
        return Caps;
    }
}
/*
 * Copyright: Andrew Gray, Full Circle Solutions
 * Date: 18th September 2014 
 */